Compute plugin accessor package name strictly from relevant files (#33519)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

## New features and usability improvements

### Kotlin build script compilation avoidance

> With this release the mechanism for detecting ABI changes in KTS build scripts gets a major overhaul.

Reviewer: ```suggestion
With this release, the mechanism for detecting ABI (Application Binary Interface) changes in [Kotlin DSL](userguide/kotlin_dsl.html) (`.kts`) build scripts has been significantly improved. 
```

Gradle's own mechanism gets replaced with the Kotlin distribution's own ABI fingerprinting.
The biggest advantage of the new mechanism is the ability to work in the presence of inline functions, something that Gradle wasn't handling efficiently until now.

> This can mean very significant improvements depending on your build and on the changes being done to the build logic.

Reviewer: ```suggestion
This results in substantial performance gains depending on your build and the changes made to the build logic.
```

> For example, in the `gradle/gradle` project, when doing non-ABI changes to build logic, configuration time is reduced by up to 60%, by avoiding a lot of build script recompilation.

Reviewer: ```suggestion
For example, in the `gradle/gradle` project, non-ABI changes to build logic now result in up to a 60% reduction in configuration time by avoiding unnecessary script recompilation.
```
